sched/scheduler/stateful_scheduler.go

```diff
@@ -92,11 +105,11 @@ func NewStatefulScheduler(
 
 	// TODO: we need to allow the scheduler to accept new jobs
 	// while recovering old ones.
-	if recoverJobsOnStartup {
+	if config.RecoverJobsOnStartup {
 		sched.startUp()
 	}
 
-	if !debugMode {
+	if !config.DebugMode {
```

Comment:
So the idea of having it in config, is we can debug real scheduler instances in prod.  We just need to implement an api, that advances the loop.  This way if we run into some nasty bug in production, we can start it up in Debug mode and Advance one step at a time to figure out what's happening.  [DX-1645](https://jira.twitter.biz/browse/DX-1645) tracks this work